### PR TITLE
Introduce `is_prefilled` to `RequestState` as equivalent to `next_start_position == 0`

### PR DESCRIPTION
**Pull request by masahi**
A follow up to https://github.com/octoml/mlc-llm/pull/134, see https://github.com/octoml/mlc-llm/pull/134#discussion_r1435260416

@elvin-n @sunggg @yelite

### CODE REVIEW

**Review by sunggg — APPROVED** [review]
LGTM, thank you! Once @elvin-n approves, I will get it merged.

**Comment by yelite** [comment]
Just for context, `next_start_position` should represent the minimum index of token that's not in the KV cache. It's added to support partial prefill (dynamic split fuse) and partial cache eviction, which is not implemented yet and probably won't be implemented in near term. 

If we want to keep `next_start_position`, it's probably better to make `is_prefilled` a `@property` derived from other fields. I can make this update later in other cleanup PR.

**Review by yelite — APPROVED** [review]

**Pull request merged**
